src/plopp/backends/matplotlib/canvas.py

```diff
@@ -174,6 +181,14 @@ def autoscale(self):
                 xmax = max(xmax, right.value)
                 ymin = min(ymin, bottom.value)
                 ymax = max(ymax, top.value)
+        if self.yscale == 'log' and ymin < 0:
+            # Something has gone wrong
+            # probably the y-limit was set before the scale changed
+            # set the y-limit to something reasonable
+            ymin = 1e-2 * ymax
+        if self.xscale == 'log' and xmin < 0:
+            xmin = 1e-2 * xmax
```

Comment:
Hmm I don't like this so much. I think we should be able to come up with a robust way of doing things where this is not needed.